src/main/java/com/mindera/pizza/domain/order/RestaurantOrder.java

```diff
@@ -92,20 +91,28 @@ public boolean addProduct(Product product) {
     }
 
     public void finishOrder() {
+        if (this.currentStatus != OrderStatus.ACCEPTED) {
+            throw new InvalidStatusChangeException(String.format(Errors.ILLEGAL_STATUS_CHANGE.toString(), this.currentStatus, OrderStatus.FINISHED));
```

Comment:
pra simplificar a chamada desta excepção, não seria mais fácil enviares aqui como parametro o Errors.ILLEGAL_STATUS_CHANGE, this.currentStatus, OrderStatus.FINISHED e depois dentro da excepção tratavas do string.format? tornava mais limpo deste lado a chamada